src/components/Api.js

```diff
@@ -0,0 +1,129 @@
+export default class Api {
+    constructor(options) {
+        // constructor body
+        this._baseURL = options.baseURL;
+        this._headers = options.headers;
+    }
+
+    getCards() {
+        return fetch(`${this._baseURL}/cards`, {
+            headers: this._headers
+        })
+        .then(res => {
+
+            // If the server returns the data okay, parse the JSON data
+            if (res.ok) {
+                return res.json();
+            }
+
+            // If the server returns an error, reject the promise
+            return Promise.reject(`Error: ${res.status}`);
+        })
+        .catch(err => {
```

Comment:
Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```

Reply:
Gotcha! That's been taken care of. Thank you.